docs/admin/healthcheck.md

```diff
@@ -0,0 +1,174 @@
+
+To troubleshoot further, you can log into the machine running Coder and attempt
+to run the following command:
+
+```shell
+curl -v ${CODER_ACCESS_URL}
+```
+
+The output of this command should aid further diagnosis.
+
+### <a name="EACS04">EACS04: /healthz did not return 200 OK</a>
+
+**Problem:** Coder was able to execute a GET request to
+`${CODER_ACCESS_URL}/healthz`, but the response code was not `200 OK` as
+expected.
+
+This could mean, for instance, that:
+
+- The request did not actually hit your Coder instance (potentially an incorrect
+  DNS entry)
+- The request hit your Coder instance, but on an unexpected path (potentially a
+  misconfigured reverse proxy)
+
+**Solution:** Inspect the `HealthzResponse` in the health check output. This
+should give you a good indication of the root cause.
+
+## Database
+
+Coder continuously executes a short database query to validate that it can reach its configured database, and also
+measures the median latency over 5 attempts.
+
+### <a name="EDB01">EDB01: Database Ping Failed</a>
+
+**Problem:** This error code is returned if any attempt to execute this database query fails.
+
+**Solution:** Investigate the health of the database.
+
+### <a name="EDB02">EDB02: Database Ping High</a>
+
+**Problem:** This code is returned if the median latency is higher than the [configured threshold](../cli/server.md#--health-check-threshold-database).
+This may not be an error as such, but is an indication of a potential issue.
+
+**Solution:** Investigate the sizing of the configured database with regard to Coder's current activity and usage. It
+may be necessary to increase the resources allocated to Coder's database. Alternatively, you can raise the configured
+threshold to a higher value (this will not address the root cause).
+
+> [!TIP]
+> - You can enable [detailed database metrics](../cli/server.md#--prometheus-collect-db-metrics) in Coder's
+> Prometheus endpoint.
+> - Fif you have [tracing enabled](../cli/server.md#--trace), these traces may also contain useful information regarding
+> Coder's database activity.
+
+## DERP
+
+Coder workspace agents may use [DERP (Designated Encrypted Relay for Packets)](https://tailscale.com/blog/how-tailscale-works/#encrypted-tcp-relays-derp) to communicate with Coder.
+This requires connectivity to a number of configured [DERP servers](../cli/server.md#--derp-config-path) which are used
+to relay traffic between Coder and workspace agents. Coder periodically queries the health of its configured DERP servers and may return one or more of the following:
+
+### <a name="EDERP01">EDERP01: DERP Node Uses Websocket</a>
+
+**Problem:** When Coder attempts to establish a connection to one or more DERP servers, it sends a specific `Upgrade: derp` HTTP header.
+Some load balancers may block this header, in which case Coder will fall back to `Upgrade: websocket`.
+This is not necessarily a fatal error, but a possible indication of a misconfigured reverse HTTP proxy.
```

Comment:
Maybe add, that workspace users may be able to reach their workspaces, but connection performance might be degraded.